apps/web/src/pages/PoolCandidates/ViewPoolCandidatePage/components/MoreActions/RevertFinalDecisionDialog.tsx

```diff
@@ -0,0 +1,228 @@
+import {
+  commonMessages,
+  formMessages,
+  getPoolCandidateStatus,
+} from "@gc-digital-talent/i18n";
+
+import Important from "./components/Important";
+
+const messages = defineMessages({
+  qualified: {
+    defaultMessage: "Qualified candidate",
+    id: "Q8ta9H",
+    description: "Message for qualified candidate",
+  },
+  disqualified: {
+    defaultMessage: "Disqualified",
+    id: "NQkEC9",
+    description: "Message for disqualified candidate",
+  },
+});
+
+const RevertFinalDecision_Mutation = graphql(/* GraphQL */ `
+  mutation RevertFinalDecision_Mutation($id: UUID!) {
+    revertFinalDecision(id: $id) {
+      id
+    }
+  }
+`);
+
+const RevertFinalDecisionDialog_Fragment = graphql(/* GraphQL */ `
+  fragment RevertFinalDecisionDialog on PoolCandidate {
+    id
+    expiryDate
+    finalDecisionAt
+    status
+  }
+`);
+
+interface RevertFinalDecisionDialogProps {
+  revertFinalDecisionQuery: FragmentType<
+    typeof RevertFinalDecisionDialog_Fragment
+  >;
+  defaultOpen?: boolean;
+}
+
+const RevertFinalDecisionDialog = ({
+  revertFinalDecisionQuery,
+  defaultOpen = false,
+}: RevertFinalDecisionDialogProps) => {
+  const intl = useIntl();
+  const [isOpen, setIsOpen] = React.useState<boolean>(defaultOpen);
+  const [, executeMutation] = useMutation(RevertFinalDecision_Mutation);
+  const { id, expiryDate, finalDecisionAt, status } = getFragment(
+    RevertFinalDecisionDialog_Fragment,
+    revertFinalDecisionQuery,
+  );
+
+  const handleError = () => {
+    toast.error(
+      intl.formatMessage({
+        defaultMessage: "Error: could not revert final decision status",
+        id: "Iv1kqA",
+        description:
+          "Message displayed when an error reverting final decision status of pool candidate",
+      }),
+    );
+  };
+
+  const handleSubmit = async () => {
+    await executeMutation({ id })
+      .then((result) => {
+        if (result.data?.revertFinalDecision) {
+          toast.success(
+            intl.formatMessage({
+              defaultMessage: "Final decision reverted successfully",
+              id: "T+fhB+",
+              description:
+                "Message displayed when a pool candidate final decision has been updated by an admin",
+            }),
+          );
+          setIsOpen(false);
+        } else {
+          handleError();
+        }
+      })
+      .catch(() => {
+        handleError();
+      });
+  };
+
+  const isQualified = () => {
+    if (
+      status === PoolCandidateStatus.QualifiedAvailable ||
+      status === PoolCandidateStatus.Expired
+    ) {
+      return true;
+    }
+
+    if (
+      status === PoolCandidateStatus.ScreenedOutApplication ||
+      status === PoolCandidateStatus.ScreenedOutAssessment
+    ) {
+      return false;
+    }
+
+    return null; // mutation should throw an error if it's any other status
+  };
+
+  const finalDecisionDate = finalDecisionAt
+    ? new Date(finalDecisionAt).toISOString().split("T")[0]
```

Comment:
We have a helper function for this now that will format and localize the date for us.  Example:

https://github.com/GCTC-NTGC/gc-digital-talent/blob/c500f21d289e6a374fd7b6945fdc0f2e1f833d82/apps/web/src/components/NotificationList/NotificationItem.tsx#L111-L117

Reply:
Fixed by [ec2f9e0](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10152/commits/ec2f9e040593ea6e0f17b4784d7c8e6d3e7c93e6)